### ISSUE TITLE
Rule `indent` should report error while `multi-line if statement` contains wrong indent.

### ISSUE THREAD

**Issue by tinymins** [eslint/eslint#14638]
**Tell us about your environment**

* **ESLint Version:**7.26.0
* **Node Version:**14
* **npm Version:**6
* **Operating System:** Ubuntu 20.04

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  rules: {
    indent: "error",
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let a = 1;

if (
    a === 1
             && a + 1 === 2
) {
    a = 3;
}
```
Online demo: https://eslint.org/demo#eyJ0ZXh0IjoibGV0IGEgPSAxO1xuXG5pZiAoXG4gICAgYSA9PT0gMVxuICAgICAgICAgICAgICYmIGEgKyAxID09PSAyXG4pIHtcbiAgICBhID0gMztcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjoyLCJmb3ItZGlyZWN0aW9uIjoyLCJnZXR0ZXItcmV0dXJuIjoyLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjoyLCJuby1jYXNlLWRlY2xhcmF0aW9ucyI6Miwibm8tY2xhc3MtYXNzaWduIjoyLCJuby1jb21wYXJlLW5lZy16ZXJvIjoyLCJuby1jb25kLWFzc2lnbiI6Miwibm8tY29uc3QtYXNzaWduIjoyLCJuby1jb25zdGFudC1jb25kaXRpb24iOjIsIm5vLWNvbnRyb2wtcmVnZXgiOjIsIm5vLWRlYnVnZ2VyIjoyLCJuby1kZWxldGUtdmFyIjoyLCJuby1kdXBlLWFyZ3MiOjIsIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6Miwibm8tZHVwZS1lbHNlLWlmIjoyLCJuby1kdXBlLWtleXMiOjIsIm5vLWR1cGxpY2F0ZS1jYXNlIjoyLCJuby1lbXB0eSI6Miwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1lbXB0eS1wYXR0ZXJuIjoyLCJuby1leC1hc3NpZ24iOjIsIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6Miwibm8tZXh0cmEtc2VtaSI6Miwibm8tZmFsbHRocm91Z2giOjIsIm5vLWZ1bmMtYXNzaWduIjoyLCJuby1nbG9iYWwtYXNzaWduIjoyLCJuby1pbXBvcnQtYXNzaWduIjoyLCJuby1pbm5lci1kZWNsYXJhdGlvbnMiOjIsIm5vLWludmFsaWQtcmVnZXhwIjoyLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6Miwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6Miwibm8tbmV3LXN5bWJvbCI6Miwibm8tb2JqLWNhbGxzIjoyLCJuby1vY3RhbCI6Miwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjoyLCJuby1yZWRlY2xhcmUiOjIsIm5vLXJlZ2V4LXNwYWNlcyI6Miwibm8tc2VsZi1hc3NpZ24iOjIsIm5vLXNldHRlci1yZXR1cm4iOjIsIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjoyLCJuby1zcGFyc2UtYXJyYXlzIjoyLCJuby10aGlzLWJlZm9yZS1zdXBlciI6Miwibm8tdW5kZWYiOjIsIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjoyLCJuby11bnJlYWNoYWJsZSI6Miwibm8tdW5zYWZlLWZpbmFsbHkiOjIsIm5vLXVuc2FmZS1uZWdhdGlvbiI6Miwibm8tdW51c2VkLWxhYmVscyI6Miwibm8tdW51c2VkLXZhcnMiOjIsIm5vLXVzZWxlc3MtY2F0Y2giOjIsIm5vLXVzZWxlc3MtZXNjYXBlIjoyLCJuby13aXRoIjoyLCJyZXF1aXJlLXlpZWxkIjoyLCJ1c2UtaXNuYW4iOjIsInZhbGlkLXR5cGVvZiI6MiwiaW5kZW50IjoyfSwiZW52Ijp7fX19

```bash
npm run lint
```

**What did you expect to happen?**
`&& a + 1 === 2` should has 2 space indent.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**
Nothing happened, check passed.

**Steps to reproduce this issue:**

1. Just use wrong indent of multiline if statement

**Are you willing to submit a pull request to fix this bug?**

Yes if I can figure it out.

**Comment by nzakas** [comment]
I believe this is working as intended. I don’t think we change indents in the middle of expressions, but could use another team member to confirm.

**Comment by mdjermanovic** [comment]
Yes, this is working as intended. The `indent` rule does not enforce indentation in the middle of expressions. This feature was discussed as an additional option (https://github.com/eslint/eslint/issues/12248).

I'm closing this issue since this isn't a bug, and per our [policies about stylistic rules](https://github.com/eslint/eslint#stylistic-rule-updates)  we won't be adding any more options to this rule. You can always make a custom rule and publish a plugin.
